azkaban-common/src/main/java/azkaban/trigger/builtin/BasicTimeChecker.java

```diff
@@ -157,8 +186,11 @@ private long calculateNextCheckTime() {
         throw new IllegalStateException(
             "100000 increments of period did not get to present time.");
       }
-      if (period == null) {
+      if (period == null && cronExpression == null) {
         break;
+      } else if (cronExecutionTime != null) {
+        Date nextDate = cronExecutionTime.getNextValidTimeAfter(date.toDate());
```

Comment:
See my earlier comment. The logic here seems to be similar to the other method I commented earlier.  Thoughts on reducing the duplication?